src/_utils.ts

```diff
@@ -84,6 +84,17 @@ export function getDefaultHost(preferPublic?: boolean) {
   return preferPublic ? "" : "localhost";
 }
 
+export function getSocketPath(name: true | string) {
+  const _name = typeof name === "string" && name.length > 0 ? name : "listhen";
+  if (platform() === "win32") {
+    if (_name.startsWith("\\\\?\\pipe\\")) {
+      return _name;
+    }
+    return `\\\\?\\pipe\\${_name}`;
+  }
+  return isAbsolute(_name) ? _name : join(tmpdir(), `${_name}.socket`);
```

Comment:
Hi and sorry to already comment on unmerged code but I really need this here so am doing all sorts of hacks to make it happen. Any reason you're forcing relative socket paths to `/tmp/`? I use relative socket paths here and it breaks. Thanks in advance!

Reply:
I've developed this feature on a linux system and have no macOS device to test it on; that's the only reason, why relative paths wasn't correctly supported. Now relative paths should also be supported on macOS. [This](https://github.com/unjs/listhen/commit/b8ef59ff6ea4afd68b7778da0887d98a2f7d75a9) issues description should clarify.

The `tmpdir()` function is returning a symbolic link on macOS, but with the `realpathSync()` function it's possible to resolve the link to the actual absolute path.

Reply:
Hi and thanks for your hard work! I'm sorry but I feel like there might have been a bit of a misunderstanding... my question was specifically about _not_ forcing relative paths into `/tmp`, instead relying on current working directory (as was my initial expectation). Is there any particular reason you force the socket into `/tmp`? Or would it be OK to simply change L95 to `return _name` instead of prepending `/tmp` and adding `.socket` ?

Reply:
I thought of a path which will be the same on all OS's. The `/tmp` directory seems to fit to this purpose. An absolute path can still be passed to through the `listhen` arguments.

Just returning the path `_name`. appending `.socket` and using the working directory is another possibility, but you can also pass the working directory as an argument. The `/tmp` directory is just used when no socket path is provided.

Reply:
I believe that's not what most people would expect :-) my use-case calls for `./frontend_run/nuxt.sock` ; I can indeed use an absolute path by manually prefixing with `$PWD` but there's not much I can do to remove the `.socket` you're adding. Why wouldn't the path be the same on all OSes if you just let it through as-is? If I request `frontend_run/nuxt.sock` I expect to get exactly that on UNIX (`$PWD/frontend_run/nuxt.sock`) and something similar on Windows (`%CD%\frontend_run\nuxt.sock`).

Reply:
Thank you for your objection!

I've changed the `getSocketPath` function to pass the path through on unixoid systems. Can't tell what was on my mind, while deciding to use the `/tmp` directory instead of passing it through to let the CWD be prepended by `createServer`...

The tests are also fixed for this new behavior.

Reply:
I think it's really good now, thanks again for all your hard work :-)